src/components/FileDetail/FileDetail.js

```diff
@@ -0,0 +1,207 @@
+import {Mutation} from 'react-apollo';
+import {GET_STUDY_BY_ID} from '../../state/queries';
+import {DELETE_FILE, FILE_DOWNLOAD_URL} from '../../state/mutations';
+import {KF_STUDY_API} from '../../common/globals';
+import TimeAgo from 'react-timeago';
+import {Button, Icon, GridContainer, Avatar} from 'kf-uikit';
+import Badge from '../Badge/Badge';
+import VersionList from '../VersionList/VersionList';
+import UploadVersionContainer from '../../containers/UploadVersionContainer';
+import {
+  fileTypeDetail,
+  formatFileSize,
+  dateCompare,
+} from '../../common/fileMeta';
+/**
+ * Form to add fields to a newly uploaded file
+ */
+
+const FileDetail = ({fileNode, onSubmit, onNameChange, history, match}) => {
+  const sortedVersions =
+    fileNode && fileNode.versions.edges.length > 0
+      ? fileNode.versions.edges.sort(dateCompare)
+      : [];
+  const latestDate =
+    sortedVersions.length > 0 ? sortedVersions[0].node.createdAt : null;
+  const latestSize =
+    sortedVersions.length > 0
+      ? formatFileSize(sortedVersions[0].node.size, true)
+      : 'unknown';
+  const studyId = match.params.kfId;
+  return (
+    <Mutation mutation={FILE_DOWNLOAD_URL}>
+      {downloadFile => (
+        <Mutation
+          mutation={DELETE_FILE}
+          update={(cache, {data: {deleteFile}}) => {
+            // Re-writes the study in the cache with the deleted file
+            // removed.
+            const {studyByKfId} = cache.readQuery({
+              query: GET_STUDY_BY_ID,
+              variables: {kfId: studyId},
+            });
+            const deleteIndex = studyByKfId.files.edges.findIndex(
+              edge => edge.node.kfId === deleteFile.kfId,
+            );
+            if (deleteIndex < 0) {
+              return studyByKfId;
+            }
+            studyByKfId.files.edges.splice(deleteIndex, 1);
+            const data = {
+              studyByKfId: {
+                ...studyByKfId,
+                files: {
+                  __typename: 'FileNodeConnection',
+                  edges: studyByKfId.files.edges,
+                },
+              },
+            };
+            cache.writeQuery({
+              query: GET_STUDY_BY_ID,
+              data,
+            });
+          }}
+        >
+          {(deleteFile, {loading, error}) => (
+            <GridContainer>
+              <Link
+                to={`/study/${match.params.kfId}/files/`}
+                className="BackButton"
+              >
+                Back to All Files
+              </Link>
+              <div className="row-2 lg:cell-10 md:cell-9 cell-12">
+                <p className="FileTitle">{fileNode.name}</p>
+                <GridContainer>
+                  <div className="cell-6 sm:cell-3">
+                    <p className="FileInfo--Title">Status:</p>
+                    <Badge state="changesRequired" />
+                  </div>
+                  <div className="cell-6 sm:cell-4">
```

Comment:
why the extra cell with `cell-4` here? seems like it's throwing off the even spacing between in this row